### PR TITLE
convert sources files to ts

### PR DESCRIPTION
**Pull request by jademcgough**
I did crap until my editor stopped giving me warnings, this can probably be improved.

noidea

### CODE REVIEW

**Comment by jademcgough** [comment]
Um, I'll fix those errors. The version issue might be trickier, since TS doesn't understand that it's provided via compilation.

**Comment by bthesorceror** [comment]
Great work Jade! I think we might also want to add types for those actions because we will need them when we convert the reducers that consume them.

File: ui/src/sources/containers/ManageSources.tsx [hunk 1]
```diff
@@ -82,33 +78,32 @@ class ManageSources extends Component {
...
-      self: string.isRequired,
-    }),
-  }),
-  sources: array,
-  notify: func.isRequired,
-  removeAndLoadSources: func.isRequired,
-  fetchKapacitors: func.isRequired,
-  setActiveKapacitor: func.isRequired,
-  deleteKapacitor: func.isRequired,
+  private handleDeleteSource = (source: Source) => () => {
```
Comment (jaredscheib) [inline ui/src/sources/containers/ManageSources.tsx]: could you add return types to both functions in this signature, even if `void`? that will prevent having to revisit this once we flip the switch on `no-implicit-any`

File: ui/src/sources/containers/ManageSources.tsx [hunk 2]
```diff
@@ -82,33 +78,32 @@ class ManageSources extends Component {
...
+
+    try {
+      this.props.removeAndLoadSources(source)
+      notify(notifySourceDeleted(source.name))
+    } catch (e) {
+      notify(notifySourceDeleteFailed(source.name))
+    }
+  }
+
+  private handleSetActiveKapacitor = ({ kapacitor }) => {
```
Comment (jaredscheib) [inline ui/src/sources/containers/ManageSources.tsx]: could you add a return type?

File: ui/src/sources/containers/ManageSources.tsx [hunk 3]
```diff
@@ -1,65 +1,61 @@
...
+  fetchKapacitors: () => void
+  setActiveKapacitor: () => void
+  deleteKapacitor: () => void
+}
 
 const V_NUMBER = VERSION // eslint-disable-line no-undef
 
 @ErrorHandling
-class ManageSources extends Component {
+class ManageSources extends Component<Props, any> {
```
Comment (jaredscheib) [inline ui/src/sources/containers/ManageSources.tsx]: if there's no `State`, do we need `any` here?

File: ui/src/sources/containers/ManageSources.tsx [hunk 4]
```diff
@@ -1,65 +1,61 @@
...
-} from 'shared/copy/notifications'
+} from 'src/shared/copy/notifications'
+
+import { Source } from 'src/types'
+
+interface Props {
+  source: Source
+  sources: Source[]
+  notify: () => void
+  remoteAndLoadSources: () => void
```
Comment (jaredscheib) [inline ui/src/sources/containers/ManageSources.tsx]: do these functions take inputs?

File: ui/src/shared/apis/index.ts [hunk 5]
```diff
@@ -237,5 +244,5 @@ export const getQueryConfigAndStatus = (url, queries, tempVars) =>
   AJAX({
     url,
     method: 'POST',
-    data: {queries, tempVars},
+    data: { queries, tempVars },
```
Comment (jaredscheib) [inline ui/src/shared/apis/index.ts]: here's an example of where `make lint` would impose our linter formatting, as mentioned above

File: ui/src/sources/containers/ManageSources.tsx [hunk 6]
```diff
@@ -1,65 +1,61 @@
-import React, {Component} from 'react'
-import PropTypes from 'prop-types'
-import {connect} from 'react-redux'
-import {bindActionCreators} from 'redux'
-import {ErrorHandling} from 'src/shared/decorators/errors'
+import React, { Component } from 'react'
```
Comment (jaredscheib) [inline ui/src/sources/containers/ManageSources.tsx]: more examples of prettier formatting :)

File: ui/src/shared/actions/sources.ts [hunk 7]
```diff
@@ -94,11 +94,13 @@ export const fetchKapacitorsAsync = source => async dispatch => {
...
   dispatch(setActiveKapacitor(kapacitor))
-  const kapacitorPost = {...kapacitor, active: true}
+  const kapacitorPost = { ...kapacitor, active: true }
   await updateKapacitorAJAX(kapacitorPost)
 }
 
-export const deleteKapacitorAsync = kapacitor => async dispatch => {
+export const deleteKapacitorAsync = (
+  kapacitor: Kapacitor,
+) => async dispatch => {
```
Comment (jaredscheib) [inline ui/src/shared/actions/sources.ts]: could you type the output of this function, even if `void`?

**Comment by 121watts** [comment]
I'm actually going to be in these files today, so I can continue this work!

**Pull request closed without merging**